src/language_servers/omnisharp.rs

```diff
@@ -0,0 +1,129 @@
+use std::fs;
+use zed_extension_api::{self as zed, settings::LspSettings, LanguageServerId, Result};
+
+pub struct Omnisharp {
+    cached_binary_path: Option<String>,
+}
+
+pub struct OmnisharpBinary {
+    pub path: String,
+    pub args: Option<Vec<String>>,
+}
+
+impl Omnisharp {
+    pub const LANGUAGE_SERVER_ID: &'static str = "omnisharp";
+
+    pub fn new() -> Self {
+        Self {
+            cached_binary_path: None,
+        }
+    }
+
+    pub fn language_server_binary(
+        &mut self,
+        language_server_id: &LanguageServerId,
+        worktree: &zed::Worktree,
+    ) -> Result<OmnisharpBinary> {
+        let binary_settings = LspSettings::for_worktree("omnisharp", worktree)
+            .ok()
+            .and_then(|lsp_settings| lsp_settings.binary);
+        let binary_args = binary_settings
+            .as_ref()
+            .and_then(|binary_settings| binary_settings.arguments.clone());
+
+        if let Some(path) = binary_settings.and_then(|binary_settings| binary_settings.path) {
+            return Ok(OmnisharpBinary {
+                path,
+                args: binary_args,
+            });
+        }
+
+        if let Some(path) = worktree.which("OmniSharp") {
+            return Ok(OmnisharpBinary {
+                path,
+                args: binary_args,
+            });
+        }
+
+        if let Some(path) = &self.cached_binary_path {
+            if fs::metadata(path).map_or(false, |stat| stat.is_file()) {
+                return Ok(OmnisharpBinary {
+                    path: path.clone(),
+                    args: binary_args,
+                });
+            }
+        }
+
+        zed::set_language_server_installation_status(
+            language_server_id,
+            &zed::LanguageServerInstallationStatus::CheckingForUpdate,
+        );
+        let release = zed::latest_github_release(
+            "OmniSharp/omnisharp-roslyn",
+            zed::GithubReleaseOptions {
+                require_assets: true,
+                pre_release: false,
+            },
+        )?;
+
+        let (platform, arch) = zed::current_platform();
+        let asset_name = format!(
+            "omnisharp-{os}-{arch}-net6.0.{extension}",
+            os = match platform {
+                zed::Os::Mac => "osx",
+                zed::Os::Linux => "linux",
```

Comment:
```suggestion
                zed::Os::Linux => "linux-gnu",
```

> https://github.com/SofusA/csharp-language-server/releases/tag/5.0.0-1.25280.9

Looks like the releases label it as `linux-gnu`